src/app/api/v1/endpoints/create_pngs.py

```diff
@@ -0,0 +1,65 @@
+"""This module defines the FastAPI endpoint for converting a PDF document to PNG images."""
+
+import os
+
+import fitz
+from app.common.aws import load_pdf_from_aws, upload_file_to_s3
+from app.common.config import config
+from app.common.schemas import PNGResponse
+from fastapi import Form, HTTPException
+
+
+def create_pngs(aws_filename: str = Form(...)):
+    """Convert a PDF document to PNG images. Please note that this function will overwrite any existing PNG files.
+
+    Args:
+        aws_filename (str): The name of the PDF document in the S3 bucket. For example, "pdfs/geoquat/train/10012.pdf".
+
+    Returns:
+        PNGResponse: The URLs of the PNG images in the S3 bucket.
+    """
+    # Validate the filename parameter
+    if not aws_filename or not isinstance(aws_filename, str):
+        raise HTTPException(
+            status_code=400, detail="Invalid request. 'filename' parameter is required and must be a string."
+        )
+
+    # Check if the PDF name is valid
+    if not aws_filename.endswith(".pdf"):
+        raise HTTPException(status_code=400, detail="Invalid request. The filename must end with '.pdf'.")
+
+    # Get the filename from the path
+    filename = aws_filename.split(".")[0]
```

Comment:
what if we have filenames that contain a `.` already before the file extension? For examples `file.one.pdf` and `file.two.pdf` ? They would both end up with `filename == 'file'`, no?

Reply:
Yes, that is certainly true. Currently, I tried to work with the filenames I observed from our data. However, would you like to proceed from here? I can also just remove the `.pdf` extension from the file, if you think that is a better solution.

Reply:
I am just not quite sure on how many different problematic filename we shall optimise for.

Reply:
I think that just removing the `.pdf` extension (and only allowing files with a `.pdf` extension in the first place) should cover all possible cases.

Reply:
Awesome! I will implement this. We were already checking for the `.pdf` extension.